added logging sub-module

de-code opened the pull request:
this implements a slightly improved version of #786

The solution mainly differs from the one outlined in #786 in the following ways:

- multiple `loggers` can be specified (still optional, and probably not needed most of the time)
- a `tqdm` class can be passed in
- Python 2 support

de-code commented on tqdm/cli.py:
```diff
@@ -1,6 +1,8 @@
 """
 Module version for monitoring CLI pipes (`... | python -m tqdm | ...`).
 """
+from __future__ import absolute_import
```
this is only required for Python 2 support, if we keep the `logging` submodule name

casperdcl reviewed (changes requested):
this is great. Perhaps should be in `tqdm.contrib.logging` or even just `tqdm.contrib.__init__` for now?

https://github.com/tqdm/tqdm/#redirecting-writing still uses `tqdm.contrib` for the `stdout/err` redirection.

de-code commented:
> this is great. Perhaps should be in `tqdm.contrib.logging` or even just `tqdm.contrib.__init__` for now?
> 
> https://github.com/tqdm/tqdm/#redirecting-writing still uses `tqdm.contrib` for the `stdout/err` redirection.

I moved it to `tqdm.contrib.logging` for now. It might be good to have a separate module just for the logging related functions.
I also moved the tests accordingly (there didn't seem to be any tests for contrib yet).
I also added a section to the README.

casperdcl commented:
fixes #786

casperdcl commented:
Hah @cthoyt it's a well overdue implementation of #786 - just never had enough free time to action it sooner; sorry. Full credit to @de-code though for one of the cleanest code PRs I've seen on this site.

user | additions | deletions
---- | --------- | ---------
@de-code | +429 | -0
@casperdcl | +102 | -170

It may look like I modded a lot but most of that was documentation, not code.
